temporal-serviceclient/src/main/java/io/temporal/serviceclient/SimpleSslContextBuilder.java

```diff
@@ -47,21 +44,49 @@
           // gRPC requires http2 protocol.
           ApplicationProtocolNames.HTTP_2);
 
+  private final PKCS pkcs;
   private final InputStream keyCertChain;
   private final InputStream key;
   private TrustManager trustManager;
   private boolean useInsecureTrustManager;
   private String keyPassword;
 
+  private enum PKCS {
+    PKCS_8,
+    PKCS_12
+  }
+
   /**
    * @param keyCertChain - an input stream for an X.509 client certificate chain in PEM format.
    * @param key - an input stream for a PKCS#8 client private key in PEM format.
+   * @deprecated use {@link #forPKCS8(InputStream, InputStream)} isntead
    */
+  @Deprecated
   public static SimpleSslContextBuilder newBuilder(InputStream keyCertChain, InputStream key) {
-    return new SimpleSslContextBuilder(keyCertChain, key);
+    return forPKCS8(keyCertChain, key);
   }
 
-  private SimpleSslContextBuilder(InputStream keyCertChain, InputStream key) {
+  /**
+   * @param keyCertChain - an input stream for an X.509 client certificate chain in PEM format.
+   * @param key - an input stream for a PKCS#8 client private key in PEM format.
+   */
+  public static SimpleSslContextBuilder forPKCS8(InputStream keyCertChain, InputStream key) {
+    return new SimpleSslContextBuilder(PKCS.PKCS_8, keyCertChain, key);
+  }
+
+  /**
+   * Note: <a href="https://github.com/temporalio/samples-server/tree/main/tls">Temporal examples
+   * for client certificates generation</a> lead to generation of Certificate Private Keys in PKCS12
+   * format.
+   *
+   * @param pfxKeyArchive - an input stream for .pfx or .p12 PKCS12 archive file
+   */
+  public static SimpleSslContextBuilder forPKCS12(InputStream pfxKeyArchive) {
```

Comment:
Can pkcs12 come with a keyCertChain?

Reply:
It can and is, but it's packaged into PFX file in this case.
This is the tricky part and why I have to do two separate methods. PKCS12 has a packaging format - PFX. And Java supports PKCS12 only packaged into PFX. You can't pass a private key and a chain separately.
PKCS8 is supported by Netty directly and there is no packaging format - so there are just two raw PEM files. One for the private key and one for cert chain.

I can try to add a separate kinda magic method that will accept just a vararg (to accommodate formats which bundle and which is not) and makes an attempt to figure things out inside, but this is a separate thing and it will not be easy..